Iterable-iOS-SDK/IterableAPI.m

```diff
@@ -612,6 +669,40 @@ - (void)trackInAppClick:(NSString *)messageId buttonURL:(NSString*)buttonURL {
     [self sendRequest:request onSuccess:[IterableAPI defaultOnSuccess:@"trackInAppClick"] onFailure:[IterableAPI defaultOnFailure:@"trackInAppClick"]];
 }
 
+/**
+ * Returns the push integration name for this app depending on the config options
+ * @return push integration name to use
+ */
+- (NSString *)pushIntegrationName {
+    if (self.config.pushIntegrationName && self.config.sandboxPushIntegrationName) {
+        switch (self.config.pushPlatform) {
+            case APNS:
+                return self.config.pushIntegrationName;
+            case APNS_SANDBOX:
+                return self.config.sandboxPushIntegrationName;
+            case AUTO:
+                return [IterableUtil isSandboxAPNS] ? self.config.sandboxPushIntegrationName : self.config.pushIntegrationName;
+        }
+    }
+    return self.config.pushIntegrationName;
```

Comment:
if self.config is nil or self.config.pushPlatform is nil we will return nil here. We should probably return the 'AUTO' case here.

Reply:
Yes, it will return nil in that case; this is by design, as there is nothing else we can return in that case.

Reply:
But registerToken:appName:pushServicePlatform:onSuccess:onFailure should check that appName is not nil.